### PR TITLE
Add an example showing how to edit a scene before it's spawned

### PR DESCRIPTION
**Pull request by mockersf**
# Objective

- Show how to edit a scene before it's spawned
- Small API improvements around `Scene`/`DynamicScene`

### CODE REVIEW

File: examples/3d/edit_gltf_scene.rs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .add_startup_system(setup)
+        .add_system(setup_scenes)
+        .run();
+}
+
+pub const CAKE_WITH_LIGHT_HANDLE: HandleUntyped = HandleUntyped::weak_from_u64(Scene::TYPE_UUID, 1);
```
Comment (alice-i-cecile) [inline examples/3d/edit_gltf_scene.rs:13]: I would add a comment explaining how these constants were chosen.

File: examples/3d/edit_gltf_scene.rs [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                .filter(|(_, name)| **name == slice_name)
+                .map(|(entity, _)| entity)
+                .next()
+                .unwrap();
+            scene_with_1_slice_missing
+                .entity_mut(first_slice_entity)
+                .despawn_recursive();
+
+            // add the modified scenes to the assets
+            scenes.set_untracked(CAKE_WITH_LIGHT_HANDLE, scene_with_lights);
```
Comment (alice-i-cecile) [inline examples/3d/edit_gltf_scene.rs:94]: set_untracked feels like a very unintuitive name for this behavior. Should it be renamed?

**Review by alice-i-cecile — CHANGES_REQUESTED** [review]
A few comments. This also needs a Migration Guide as there are breaking changes :)

File: examples/3d/edit_gltf_scene.rs (hunk 1 above)
Comment (mockersf) [reply examples/3d/edit_gltf_scene.rs:13]: would you believe me if I said "randomly selected value"? 😄

File: examples/3d/edit_gltf_scene.rs (hunk 2 above)
Comment (mockersf) [reply examples/3d/edit_gltf_scene.rs:94]: `set_untracked` creates an asset that is not tracked for automatic cleanup, the name seems right?

Comment (alice-i-cecile) [reply examples/3d/edit_gltf_scene.rs:94]: Why not `create_untracked` then? `set` to me implies that it's setting an existing asset.

File: examples/3d/edit_gltf_scene.rs (hunk 1 above)
Comment (alice-i-cecile) [reply examples/3d/edit_gltf_scene.rs:13]: *grin* Of course ;) Good to explain to the users though.

**Comment by mockersf** [comment]
converting to draft as most comments are related to how to do asset post processing, and it's just not something that can be pretty in Bevy for now

**Pull request closed without merging**